pkg/cluster/delete.go

```diff
@@ -443,7 +444,7 @@ func (m *manager) Delete(ctx context.Context) error {
 		return err
 	}
 
-	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil && m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil {
+	if utilwimi.IsWimi(m.doc.OpenShiftCluster) {
```

Comment:
Just a small comment: If we move the definition of the IsWimi function to the OpenShift cluster, we obtain a richer domain where functions are closer where the data is used (to avoid having an anemic Domain Model, a concept of DDD that here could be interesting). 
If doing that (check next comments) that line would look like:
```go
if m.doc.OpenShiftCluster.IsWimi() {
```
which is pretty neat as clearly states that the fact that a cluster is Wimi or not just depend on the cluster and reads a bit better 🙂